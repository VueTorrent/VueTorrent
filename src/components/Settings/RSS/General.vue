<script setup lang="ts">

import { usePreferenceStore, useVueTorrentStore } from '@/stores'

const preferenceStore = usePreferenceStore()
const vuetorrentStore = useVueTorrentStore()
</script>

<template>
  <v-list>
    <v-list-subheader>{{ $t('settings.rss.general.reader.subheader') }}</v-list-subheader>

    <v-list-item>
      <v-checkbox v-model="preferenceStore.preferences!.rss_processing_enabled" hide-details
                  :label="$t('settings.rss.general.reader.enableProcessing')" />

      <v-row>
        <v-col cols="12" sm="6">
          <v-text-field
            v-model="preferenceStore.preferences!.rss_refresh_interval"
            type="number"
            hide-details
            :suffix="$t('units.minutes')"
            :label="$t('settings.rss.general.reader.feedsRefreshInterval')" />
        </v-col>
        <v-col cols="12" sm="6">
          <v-text-field v-model="preferenceStore.preferences!.rss_max_articles_per_feed" type="number"
                        :label="$t('settings.rss.general.reader.maximumArticlesPerFeed')" />
        </v-col>
      </v-row>
    </v-list-item>

    <v-divider />
    <v-list-subheader>{{ $t('settings.rss.general.autoDownloader.subheader') }}</v-list-subheader>

    <v-list-item>
      <v-checkbox v-model="preferenceStore.preferences!.rss_auto_downloading_enabled" hide-details class="ma-0 pa-0"
                  :label="$t('settings.rss.general.autoDownloader.enable')" />
    </v-list-item>

    <v-divider />
    <v-list-subheader>{{ $t('settings.rss.general.smartEpisodeFilter.subheader') }}</v-list-subheader>

    <v-list-item>
      <v-checkbox
        v-model="preferenceStore.preferences!.rss_download_repack_proper_episodes"
        hide-details
        class="ma-0 pa-0"
        :label="$t('settings.rss.general.smartEpisodeFilter.downloadEpisodes')" />
      <v-textarea
        v-model="preferenceStore.preferences!.rss_smart_episode_filters"
        outlined
        auto-grow
        clearable
        persistent-hint
        :hint="$t('settings.rss.general.smartEpisodeFilter.filtersHint')"
        :label="$t('settings.rss.general.smartEpisodeFilter.filters')" />
    </v-list-item>

    <v-divider />

    <v-list-item>
      <v-checkbox v-model="vuetorrentStore.useIdForRssLinks"
                  hide-details
                  :label="$t('settings.rss.general.useIdForRssLinks')" />
    </v-list-item>
  </v-list>
</template>

<style scoped lang="scss"></style>
