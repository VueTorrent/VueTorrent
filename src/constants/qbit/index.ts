import * as AppPreferences from './AppPreferences'
import { ConnectionStatus } from './ConnectionStatus'
import { FilePriority } from './FilePriority'
import { FilterState } from './FilterState'
import { LogType } from './LogType'
import { PieceState } from './PieceState'
import { SortOptions } from './SortOptions'
import { TorrentOperatingMode } from './TorrentOperatingMode'
import { TorrentState } from './TorrentState'
import { TrackerStatus } from './TrackerStatus'

export {
  AppPreferences,
  ConnectionStatus,
  FilterState,
  LogType,
  PieceState,
  FilePriority,
  SortOptions,
  TrackerStatus,
  TorrentOperatingMode,
  TorrentState
}
