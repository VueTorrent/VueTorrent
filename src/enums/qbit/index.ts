import * as AppPreferences from './AppPreferences'
import { LogType } from './LogType'
import { Priority } from './Priority'
import { TrackerStatus } from './TrackerStatus'
import { ConnectionStatus } from './ConnectionStatus'
import { TorrentState } from './TorrentState'

export { AppPreferences, ConnectionStatus, LogType, Priority, TrackerStatus, TorrentState }
