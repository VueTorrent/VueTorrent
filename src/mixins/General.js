import { v4 as uuidv4 } from 'uuid'
import { mapGetters } from 'vuex'
export default {
    computed: {
        ...mapGetters(['getTheme']),
        theme() {
            return this.getTheme() ? 'dark' : 'light'
        },
        background() {
            return this.$vuetify.theme.themes[this.theme].background
        }
    },
    methods: {
        createModal(name, props) {
            const component = {
                component: name,
                props,
                guid: uuidv4()
            }

            this.$store.commit('ADD_MODAL', component)
        }
    }
}
