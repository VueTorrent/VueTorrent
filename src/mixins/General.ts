import { v4 as uuidv4 } from 'uuid'
import { mapGetters } from 'vuex'
import { defineComponent } from 'vue'

export default defineComponent({
  computed: {
    ...mapGetters(['getTheme']),
    theme() {
      // @ts-expect-error: TS2339: Property 'getTheme' does not exist on type 'CreateComponentPublicInstance{}, unknown, {}, {}, {}, ComponentOptionsMixin, ComponentOptionsMixin, {}, {}, {}, false, OptionTypesType{}, {}, {}, {}, {}, {}>, ... 5 more ..., {}>'.
      return this.getTheme()
    },
    isMobile() {
      return this.$vuetify.breakpoint.smAndDown
    }
  },
  methods: {
    createModal(name: string, props?: any) {
      const component = {
        component: name,
        props,
        guid: uuidv4()
      }

      this.$store.commit('ADD_MODAL', component)
    }
  }
})
