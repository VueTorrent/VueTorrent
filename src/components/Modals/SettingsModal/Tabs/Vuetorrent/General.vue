<template>
  <v-card flat>
    <v-subheader>
      These settings are for the custom WebUI
      itself
    </v-subheader>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="showCurrentSpeed"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Show Current Speed"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="showSpeedGraph"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Show Speed Graph"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="showSessionStat"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Show Session Stats"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="freeSpace"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Show Free Space"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="showTrackerFilter"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Show Tracker Filter"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col>
          <v-switch
            v-model="useDeviceDarkMode"
            class="v-input--reverse pa-0 ma-0"
            hide-details
            inset
            label="Use device dark/light mode"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col cols="8" sm="8" md="10">
          <p class="subtitle-1 mt-2">
            Pagination size:
          </p>
        </v-col>
        <v-col cols="4" sm="4" md="2">
          <v-select
            v-model="paginationSize"
            flat
            solo
            dense
            hide-details
            background-color="background"
            class="rounded-xl"
            :items="paginationSizes"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col cols="8" sm="8" md="10">
          <p class="subtitle-1 mt-2">
            VueTorrent title:
          </p>
        </v-col>
        <v-col cols="4" sm="4" md="2">
          <v-select
            v-model="title"
            flat
            solo
            dense
            hide-details
            background-color="background"
            class="rounded-xl"
            :items="titleOptions"
          />
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col cols="10" sm="10" md="11">
          <p class="subtitle-1">
            Current Version:
          </p>
        </v-col>
        <v-col cols="2" sm="2" md="1">
          <a target="_blank" :href="`https://github.com/WDaan/VueTorrent/releases/tag/v${version}`">
            <p class="mb-2">{{ version }}</p>
          </a>
        </v-col>
      </v-row>
    </v-list-item>
    <v-list-item>
      <v-row dense>
        <v-col cols="10" sm="10" md="11">
          <p class="subtitle-1">
            QBittorrent Version:
          </p>
        </v-col>
        <v-col cols="2" sm="2" md="1">
          <a target="_blank" :href="`https://github.com/qbittorrent/qBittorrent/releases/tag/release-${Qbitversion}`">
            <p class="mb-2">
              {{ Qbitversion }}
            </p>
          </a>
        </v-col>
      </v-row>
    </v-list-item>
  </v-card>
</template>

<script>
import { mapState, mapGetters } from 'vuex'
import qbit from '@/services/qbit'
export default {
  name: 'General',
  data() {
    return {
      paginationSizes: [5, 15, 30, 50],
      titleOptions: ['Default', 'Global Speed', 'First Torrent Status'],
      Qbitversion: 0
    }
  },
  computed: {
    ...mapState(['webuiSettings']),
    ...mapGetters(['getAppVersion']),
    freeSpace: {
      get() {
        return this.webuiSettings.showFreeSpace
      },
      set(val) {
        this.webuiSettings.showFreeSpace = val
      }
    },
    showCurrentSpeed: {
      get() {
        return this.webuiSettings.showCurrentSpeed
      },
      set(val) {
        this.webuiSettings.showCurrentSpeed = val
      }
    },
    showSpeedGraph: {
      get() {
        return this.webuiSettings.showSpeedGraph
      },
      set(val) {
        this.webuiSettings.showSpeedGraph = val
      }
    },
    showSessionStat: {
      get() {
        return this.webuiSettings.showSessionStat
      },
      set(val) {
        this.webuiSettings.showSessionStat = val
      }
    },
    showTrackerFilter: {
      get() {
        return this.webuiSettings.showTrackerFilter
      },
      set(val) {
        this.webuiSettings.showTrackerFilter = val
      }
    },
    useDeviceDarkMode: {
      get() {
        return this.webuiSettings.useDeviceDarkMode
      },
      set(val) {
        this.webuiSettings.useDeviceDarkMode = val
      }
    },
    paginationSize: {
      get() {
        return this.webuiSettings.paginationSize
      },
      set(val) {
        this.webuiSettings.paginationSize = val
      }
    },
    title: {
      get() {
        return this.webuiSettings.title
      },
      set(val) {
        this.webuiSettings.title = val
      }
    },
    version() {
      return this.getAppVersion()
    }
  },
  created() {
    this.fetchQbitVersion()
  },
  methods: {
    async fetchQbitVersion() {
      const version = await qbit.getAppVersion()
      this.Qbitversion = version.includes('v') ? version.substring(1) : version
    }
  }
}
</script>

<style lang="scss" scoped>
// Reversed input variant
::v-deep .v-input--reverse .v-input__slot {
  flex-direction: row-reverse;
  justify-content: flex-end;
  .v-application--is-ltr & {
    .v-input--selection-controls__input {
      margin-right: 0;
      margin-left: 8px;
    }
  }
  .v-application--is-rtl & {
    .v-input--selection-controls__input {
      margin-left: 0;
      margin-right: 8px;
    }
  }
}
</style>