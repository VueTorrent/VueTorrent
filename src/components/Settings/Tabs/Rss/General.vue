<template>
  <v-card flat>
    <v-subheader>
      {{ $t('modals.settings.pageRss.pageGeneral.rssAutoProcessing') }}
    </v-subheader>
    <v-list-item>
      <v-checkbox v-model="settings.rss_processing_enabled" hide-details class="ma-0 pa-0" :label="$t('modals.settings.pageRss.pageGeneral.input.enableRssProcessing')" />
    </v-list-item>
    <v-list-item>
      <v-text-field
        v-model="settings.rss_refresh_interval"
        class="mb-2"
        outlined
        dense
        type="number"
        hide-details
        :label="$t('modals.settings.pageRss.pageGeneral.input.feedsRefreshInterval')"
        :disabled="!settings.rss_processing_enabled"
      />
    </v-list-item>
    <v-list-item>
      <v-text-field
        v-model="settings.rss_max_articles_per_feed"
        class="mb-2"
        outlined
        dense
        type="number"
        hide-details
        :label="$t('modals.settings.pageRss.pageGeneral.input.feedsMaxArticles')"
        :disabled="!settings.rss_processing_enabled"
      />
    </v-list-item>
    <v-divider />
    <v-subheader>
      {{ $t('modals.settings.pageRss.pageGeneral.rssAutoDownloader') }}
    </v-subheader>
    <v-list-item>
      <v-checkbox v-model="settings.rss_auto_downloading_enabled" hide-details class="ma-0 pa-0" :label="$t('modals.settings.pageRss.pageGeneral.input.enableRssAutoDownload')" />
    </v-list-item>
  </v-card>
</template>
<script>
import { FullScreenModal, SettingsTab } from '@/mixins'

export default {
  name: 'RssGeneral',
  mixins: [SettingsTab, FullScreenModal]
}
</script>

<style lang="scss" scoped>
:deep(.v-select .v-select__selection) {
  padding-top: 10px !important;
  padding-bottom: 10px !important;
}
// Reversed input variant
:deep(.v-input--reverse .v-input__slot) {
  @import '../../../../styles/styles';
  @include reverse-switch;
}
</style>
