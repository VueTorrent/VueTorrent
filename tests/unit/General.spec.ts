import { describe, it, expect, beforeEach, vi } from 'vitest'
import { shallowMount, Wrapper } from '@vue/test-utils'
import General from '@/components/Settings/Tabs/VueTorrent/VGeneral.vue'
import { Qbit } from '../../src/services/qbit'

let wrapper: Wrapper<any>

vi.mock('../../src/services/qbit', () => {
  return {
    Qbit: {
      getAppVersion: () => Promise.resolve('v1.2.3')
    }
  }
})

describe('General', () => {
  beforeEach(() => {
    wrapper = shallowMount(General, {
      mocks: {
        $t: () => {},
        $store: {
          getters: { getAppVersion: vi.fn() },
          state: {
            webuiSettings: {
              showCurrentSpeed: 100,
              showSpeedGraph: true,
              showSessionStat: true,
              showAlltimeStat: true,
              showTrackerFilter: true,
              rightDrawer: 10,
              paginationSize: 10,
              title: 'test',
              lang: 'English'
            }
          }
        }
      }
    })
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('tests showCurrentSpeed', () => {
    expect(wrapper.vm.showCurrentSpeed).toEqual(100)
  })

  it('tests showSpeedGraph', () => {
    expect(wrapper.vm.showSpeedGraph).toEqual(true)
  })

  it('tests showSessionStat', () => {
    expect(wrapper.vm.showSessionStat).toEqual(true)
  })

  it('tests showAlltimeStat', () => {
    expect(wrapper.vm.showAlltimeStat).toEqual(true)
  })

  it('tests showTrackerFilter', () => {
    expect(wrapper.vm.showTrackerFilter).toEqual(true)
  })

  it('tests rightDrawer', () => {
    expect(wrapper.vm.rightDrawer).toEqual(10)
  })

  it('tests paginationSize', () => {
    expect(wrapper.vm.paginationSize).toEqual(10)
  })

  it('tests title', () => {
    expect(wrapper.vm.title).toEqual('test')
  })

  it('tests lang', () => {
    expect(wrapper.vm.lang).toEqual('English')
  })
})
