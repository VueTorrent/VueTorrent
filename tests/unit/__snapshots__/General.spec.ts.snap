// Vitest Snapshot v1

exports[`General > render correctly 1`] = `
"<v-card-stub data-v-7da6d3e2=\\"\\" loaderheight=\\"4\\" tag=\\"div\\" flat=\\"true\\">
  <v-subheader-stub data-v-7da6d3e2=\\"\\"> </v-subheader-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"100\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"true\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"true\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"true\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"true\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inputvalue=\\"10\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\">
        <v-switch-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" hidedetails=\\"true\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" inset=\\"true\\" class=\\"v-input--reverse pa-0 ma-0\\"></v-switch-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"8\\" sm=\\"8\\" md=\\"10\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1 mt-2\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"4\\" sm=\\"4\\" md=\\"2\\" tag=\\"div\\">
        <v-select-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"English\\" appendicon=\\"$dropdown\\" backgroundcolor=\\"background\\" dense=\\"true\\" hidedetails=\\"true\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" flat=\\"true\\" solo=\\"true\\" type=\\"text\\" valuecomparator=\\"[Function]\\" nodatatext=\\"$vuetify.noDataText\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" itemcolor=\\"primary\\" itemdisabled=\\"disabled\\" itemtext=\\"caption\\" itemvalue=\\"value\\" menuprops=\\"[object Object]\\" class=\\"rounded-xl\\"></v-select-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"8\\" sm=\\"8\\" md=\\"10\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1 mt-2\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"4\\" sm=\\"4\\" md=\\"2\\" tag=\\"div\\">
        <v-select-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"10\\" appendicon=\\"$dropdown\\" backgroundcolor=\\"background\\" dense=\\"true\\" hidedetails=\\"true\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" flat=\\"true\\" solo=\\"true\\" type=\\"text\\" valuecomparator=\\"[Function]\\" nodatatext=\\"$vuetify.noDataText\\" items=\\"5,15,30,50\\" itemcolor=\\"primary\\" itemdisabled=\\"disabled\\" itemtext=\\"text\\" itemvalue=\\"value\\" menuprops=\\"[object Object]\\" class=\\"rounded-xl\\"></v-select-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"8\\" sm=\\"8\\" md=\\"10\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1 mt-2\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"4\\" sm=\\"4\\" md=\\"2\\" tag=\\"div\\">
        <v-select-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"test\\" appendicon=\\"$dropdown\\" backgroundcolor=\\"background\\" dense=\\"true\\" hidedetails=\\"true\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" flat=\\"true\\" solo=\\"true\\" type=\\"text\\" valuecomparator=\\"[Function]\\" nodatatext=\\"$vuetify.noDataText\\" items=\\"Default,Global Speed,First Torrent Status\\" itemcolor=\\"primary\\" itemdisabled=\\"disabled\\" itemtext=\\"text\\" itemvalue=\\"value\\" menuprops=\\"[object Object]\\" class=\\"rounded-xl\\"></v-select-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" sm=\\"6\\" md=\\"9\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1 mt-2\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" sm=\\"6\\" md=\\"3\\" tag=\\"div\\">
        <v-text-field-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" dense=\\"true\\" hidedetails=\\"true\\" label=\\"using Dayjs\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" outlined=\\"true\\" placeholder=\\"DD/MM/YYYY, HH:mm:ss\\" type=\\"text\\" class=\\"mb-2\\"></v-text-field-stub>
      </v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"10\\" sm=\\"10\\" md=\\"11\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"2\\" sm=\\"2\\" md=\\"1\\" tag=\\"div\\"><a data-v-7da6d3e2=\\"\\" target=\\"_blank\\" href=\\"https://github.com/WDaan/VueTorrent/releases/tag/vundefined\\">
          <p data-v-7da6d3e2=\\"\\" class=\\"mb-2\\"></p>
        </a></v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-row-stub data-v-7da6d3e2=\\"\\" tag=\\"div\\" dense=\\"true\\">
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"10\\" sm=\\"10\\" md=\\"11\\" tag=\\"div\\">
        <p data-v-7da6d3e2=\\"\\" class=\\"subtitle-1\\"> </p>
      </v-col-stub>
      <v-col-stub data-v-7da6d3e2=\\"\\" cols=\\"2\\" sm=\\"2\\" md=\\"1\\" tag=\\"div\\"><a data-v-7da6d3e2=\\"\\" target=\\"_blank\\" href=\\"https://github.com/qbittorrent/qBittorrent/releases/tag/release-v1.2.3\\">
          <p data-v-7da6d3e2=\\"\\" class=\\"mb-2\\"> v1.2.3 </p>
        </a></v-col-stub>
    </v-row-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\">
    <v-textarea-stub data-v-7da6d3e2=\\"\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"\\" backgroundcolor=\\"\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\" rowheight=\\"24\\" rows=\\"5\\"></v-textarea-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\" class=\\"remove-after justify-content-evenly\\">
    <v-btn-stub data-v-7da6d3e2=\\"\\" tag=\\"button\\" activeclass=\\"\\" type=\\"button\\"></v-btn-stub>
    <v-btn-stub data-v-7da6d3e2=\\"\\" tag=\\"button\\" activeclass=\\"\\" type=\\"button\\"></v-btn-stub>
  </v-list-item-stub>
  <v-list-item-stub data-v-7da6d3e2=\\"\\" activeclass=\\"\\" tag=\\"div\\" class=\\"justify-center pb-2\\">
    <v-btn-stub data-v-7da6d3e2=\\"\\" color=\\"red\\" dark=\\"true\\" tag=\\"button\\" activeclass=\\"\\" type=\\"button\\"></v-btn-stub>
  </v-list-item-stub>
</v-card-stub>"
`;
